src/Microsoft.AspNet.Http/ReadableStringCollection.cs

```diff
@@ -13,7 +13,15 @@ namespace Microsoft.AspNet.Http.Internal
     /// </summary>
     public class ReadableStringCollection : IReadableStringCollection
     {
-        public static readonly IReadableStringCollection Empty = new ReadableStringCollection(new Dictionary<string, StringValues>(0));
+        public static readonly IReadableStringCollection Empty = new ReadableStringCollection();
+
+        private static readonly string[] EmptyKeys = new string[0];
+        private static readonly IEnumerable<KeyValuePair<string, StringValues>> EmptyEnumerable = (IEnumerable<KeyValuePair<string, StringValues>>)(new KeyValuePair<string, StringValues>[0]);
```

Comment:
Not sure where that is? Can't seem to grab a ref.

Can do the ```#if !DNXCORE50``` stuff on ```EmptyKeys``` with ```Array.Empty<string>()``` though?
Also ```#if DNXCORE50``` for ```EmptyEnumerable = Array.Empty<KeyValuePair<string, StringValues>>())```

Didn't know if it was too messy